GooglePlayInstant/Editor/PlayInstantQuickDeployHttpHandler.cs

```diff
@@ -0,0 +1,361 @@
+        /// Sends a general POST request to the provided endpoint, along with the data provided in the byte array and header
+        /// parameters
+        /// </summary>
+        /// <param name="endpoint">Endpoint to which the data should be sent </param>
+        /// <param name="postData">An array of bytes representing the data that will be passed in the POST request body</param>
+        /// <param name="postHeaders"> A dictionary representing a set of key-value pairs to be added to the request headers</param>
+        /// <returns></returns>
+        public static string SendHttpPostRequest(string endpoint, byte[] postData,
+            Dictionary<string, string> postHeaders)
+        {
+            var form = new WWWForm();
+            if (postHeaders != null)
+            {
+                foreach (var pair in postHeaders)
+                {
+                    form.headers.Add(pair.Key, pair.Value);
+                }
+            }
+
+            var www = new WWW(endpoint, postData, form.headers);
+            return GetResultWhenDone(www);
+        }
+
+        /// <summary>
+        /// Sends a general POST request to the provided endpoint, along with the data provided in the form and headers
+        /// parameters
+        /// </summary>
+        /// <param name="endpoint">Endpoint to which the data should be sent </param>
+        /// <param name="postForm">A dictionary representing a set of key-value paris to be added to the request body</param>
+        /// <param name="postHeaders"> A dictionary representing a set of key-value pairs to be added to the request headers</param>
+        /// <returns></returns>
+        public static string SendHttpPostRequest(string endpoint, Dictionary<string, string> postForm,
+            Dictionary<string, string> postHeaders)
+        {
+            var form = new WWWForm();
+            if (postForm != null)
+            {
+                foreach (var pair in postForm)
+                {
+                    form.AddField(pair.Key, pair.Value);
+                }
+            }
+
+            return SendHttpPostRequest(endpoint, postForm != null ? form.data : null, postHeaders);
+        }
+
+        /// <summary>
+        /// Sends a general GET request to the specified endpoint along with specified parameters and headers
+        /// </summary>
+        /// <param name="endpoint">The endpoint to with the get request should be sent</param>
+        /// <param name="getParams"> A dictionary representing ke-value pairs that should be attached to the endpoint as GET parameters</param>
+        /// <param name="getHeaders">A dictionary representing key-value pairs that constitude the data to be added to the request headers</param>
+        /// <returns></returns>
+        public static string SendHttpGetRequest(string endpoint, Dictionary<string, string> getParams,
+            Dictionary<string, string> getHeaders)
+        {
+            var fullEndpoint = endpoint + "?";
```

Comment:
I can't believe that .NET doesn't have proper query string builder. :(

There are a few issues with this custom approach (this line and loop below):
- The "?" should only be added if there is at least one getParams pair
- There should be no trailing "&" after the last pair
- Some characters in a URL need to be URL encoded/escaped

Instead, how about something like this

```
var builder = new UriBuilder(endpoint);
// Assume Query is empty to start with
builder.Query = string.Join("&",
    getParams.Select(kvp => string.Format("{0}={1}", WWW.EscapeURL(kvp.Key), WWW.EscapeURL(kvp.Value))).ToArray());
var fullEndpoint = builder.ToString();
```